src/go/k8s/cmd/configurator/main.go

```diff
@@ -174,28 +218,65 @@ func registerAdvertisedKafkaAPI(
 		return nil
 	}
 
-	k8sconfig, err := rest.InClusterConfig()
+	node, err := getNode(c.nodeName)
 	if err != nil {
-		return fmt.Errorf("unable to create in cluster config: %w", err)
+		return fmt.Errorf("unable to retrieve node: %w", err)
 	}
 
-	clientset, err := kubernetes.NewForConfig(k8sconfig)
-	if err != nil {
-		return fmt.Errorf("unable to create clientset: %w", err)
+	cfg.Redpanda.AdvertisedKafkaApi = append(cfg.Redpanda.AdvertisedKafkaApi, config.NamedSocketAddress{
+		SocketAddress: config.SocketAddress{
+			Address: getExternalIP(node),
+			Port:    c.hostPort,
+		},
+		Name: "kafka-external",
+	})
+
+	return nil
+}
+
+// nolint:dupl // todo
+func registerAdvertisedPandaproxyAPI(
+	c *configuratorConfig, cfg *config.Config, index brokerID, proxyAPIPort int,
+) error {
+	cfg.Pandaproxy.AdvertisedPandaproxyAPI = []config.NamedSocketAddress{
+		{
+			SocketAddress: config.SocketAddress{
+				Address: c.hostName + "." + c.svcFQDN,
+				Port:    proxyAPIPort,
+			},
+			Name: "proxy",
+		},
+	}
+
+	if !c.proxyExternalConnectivity {
+		return nil
+	}
+
+	// Pandaproxy uses the Kafka API subdomain.
+	if len(c.subdomain) > 0 {
```

Comment:
what if pandaproxy does not specify subdomain but kafka does? that's also not allowed? and is it the case for admin api as well?